cmd/really/really.cpp

```diff
@@ -141,38 +142,36 @@ class ReallyServerDelegate : public quicr::ServerDelegate
     server = std::move(server_in);
   }
 
-  virtual void onPublishIntent(const quicr::Namespace& quicr_namespace,
-                               const std::string& /* origin_url */,
-                               bool /* use_reliable_transport */,
-                               const std::string& /* auth_token */,
-                               quicr::bytes&& /* e2e_token */)
+  void onPublishIntent(const quicr::Namespace& quicr_namespace,
+                       const std::string& /* origin_url */,
+                       bool /* use_reliable_transport */,
+                       const std::string& /* auth_token */,
+                       quicr::bytes&& /* e2e_token */) override
   {
-    // TODO: Authenticate token
+    // TODO(trigaux): Authenticate token
     logger->info << "Publish intent namespace: " << quicr_namespace
                  << std::flush;
 
     // TODO(trigaux): Move logic into quicr::Server
-    quicr::PublishIntentResult result{ quicr::messages::Response::Ok, {}, {} };
+    const auto result =
+      quicr::PublishIntentResult{ quicr::messages::Response::Ok, {}, {} };
     server->publishIntentResponse(quicr_namespace, result);
   };
 
-  virtual void onPublishIntentEnd(const quicr::Namespace& /* quicr_namespace */,
-                                  const std::string& /* auth_token */,
-                                  quicr::bytes&& /* e2e_token */)
+  void onPublishIntentEnd(const quicr::Namespace& /* quicr_namespace */,
+                          const std::string& /* auth_token */,
+                          quicr::bytes&& /* e2e_token */) override
   {
   }
 
-  virtual void onPublisherObject(
-    const qtransport::TransportContextId& context_id,
-    [[maybe_unused]] const qtransport::StreamId& stream_id,
-    [[maybe_unused]] bool use_reliable_transport,
-    quicr::messages::PublishDatagram&& datagram)
+  void onPublisherObject(const qtransport::TransportContextId& context_id,
+                         [[maybe_unused]] const qtransport::StreamId& stream_id,
+                         [[maybe_unused]] bool use_reliable_transport,
+                         quicr::messages::PublishDatagram&& datagram) override
   {
-    std::list<Subscriptions::Remote> list =
-      subscribeList.find(datagram.header.name);
+    const auto list = subscribeList.find(datagram.header.name);
```

Comment:
IMO, use of auto is "bad". The keyword IMO is a somewhat bad introduction in C++, anyway. I have abused it where I was too lazy to specify the type, but I definitely would not purposely change the code like this.  If there is a type change, this would catch it.  That might be good or bad, but I would argue it is generally better to be explicit.  One must use `auto` on lamdas and there are some functions where `auto` is necessary since the input type might be a plurality of things.  So while there are valid uses for it, I would limit use to those cases where it's truly the best choice.

Reply:
It's not about laziness.  See [Almost Always Auto](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/)

Reply:
This is like taking a step toward loose typing. I totally disagree. I could write a blog post on why `auto` should be considered harmful. It's just bad. It makes sense for some situations, but it most definitely should not be used almost always.

Reply:
Every modern programming language disagrees with you, even the ones with very strong type systems (some far stronger than C++).  Go has `:=`, Rust has `let`, Swift has `let` and `var`, Kotlin has `val` and `var`, ... even Java has `var`.  Type inference is the modern way.

Reply:
I think we all need to start writing in Ada.

Reply:
More seriously, though, `auto foo = something->second` just makes reading code harder since one has to go hunt down what `second` is. I've run into the same thing in Python.  It's just bad for managing code. It works, but bad. I don't care if other languages want to create the same bad practice.